cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -325,6 +329,9 @@ public synchronized ProgramController start(ProgramId programId, Map<String, Str
     final ProgramController controller = runtimeInfo.getController();
     final String runId = controller.getRunId().getId();
     final String twillRunId = runtimeInfo.getTwillRunId() == null ? null : runtimeInfo.getTwillRunId().getId();
+    final RetryStrategy retryStrategy = SystemArguments.getRetryStrategy(userArguments.asMap(), programId.getType(),
```

Comment:
hm I see. I think we can just retry for some constant amount of time then, it shouldn't be dependent on the program type.